### PR TITLE
DDFFORM 313

### PR DESCRIPTION
**Pull request by LasseStaus**
#### Link to issue

[DDFFORM-313](https://reload.atlassian.net/browse/DDFFORM-313)

Other relevant PRs: 

- React PR: https://github.com/danskernesdigitalebibliotek/dpl-react/pull/971
- Design system PR: https://github.com/danskernesdigitalebibliotek/dpl-design-system/pull/517 

#### Description

This PR adds a paragraph for the react app material-grid-automatic.  

The PR also creates a custom module, DPL_FBI, which will be used for handling the logic of retrieving data from FBI in relation to fields, as well as creating widgets for the editorial experience later on. 

As a start, this also creates a custom field; cql_search. 

#### Screenshot of the result

image

#### Additional comments or questions

Please consider the `"mnsami/composer-custom-directory-installer": true` added in composer. I'm not completely sure if this is necessary. 

Other than that comment on anything you find.

### CODE REVIEW

File: web/modules/custom/dpl_fbi/src/Plugin/Field/FieldType/CqlSearchItem.php [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      ->setLabel(t('Text value'))
+      ->setRequired(TRUE);
+
+    return $properties;
+  }
+
+  /**
+   * Getting the constraints for the field.
+   *
+   * @return array<mixed>
```
Comment (kasperg) [inline web/modules/custom/dpl_fbi/src/Plugin/Field/FieldType/CqlSearchItem.php:54]: :+1: I see that you resolved the static analysis errors whatever they were without resolving to a global ignore.

File: web/modules/custom/dpl_fbi/src/Plugin/Field/FieldType/CqlSearchItem.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+namespace Drupal\dpl_fbi\Plugin\Field\FieldType;
+
+use Drupal\Component\Utility\Random;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\TypedData\DataDefinition;
+
+/**
+ * Defines the 'dpl_fbi_cql_search' field type.
```
Comment (kasperg) [inline web/modules/custom/dpl_fbi/src/Plugin/Field/FieldType/CqlSearchItem.php:14]: Consider if you can provide more information about what this field represents.

**Review by kasperg — CHANGES_REQUESTED** [review]
:+1: Well done. I think this is well handled. 

My quips are mostly regarding suggestions for improving the generated code for the field to match our use case.

File: composer.json [hunk 3]
```diff
@@ -162,7 +162,8 @@
             "drupal/core-project-message": true,
             "phpstan/extension-installer": true,
             "zaporylie/composer-drupal-optimizations": true,
-            "drupal/console-extend-plugin": true
+            "drupal/console-extend-plugin": true,
+            "mnsami/composer-custom-directory-installer": true
```
[the remark this answers is not in the document]
Comment (LasseStaus) [reply composer.json:166]: Makes sense, removed it.

**Comment by LasseStaus** [comment]
Superseded by #826

**Pull request closed without merging**